pypesto/sampling/diagnostics.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from ..result import Result
+from .geweke_test import burn_in_by_sequential_geweke
+from .auto_correlation import auto_correlation
+
+
+def geweke_test(result: Result,
+                zscore: float = 2.):
+    """
+    Calculates the burn-in of MCMC chains.
+
+    Parameters
+    ----------
+    result:
+        The pyPESTO result object with filled sample result.
+    zscore:
+        The Geweke test threshold. Default 2.
+
+    Returns
+    -------
+    burn_in:
+        Iteration where the first and the last fraction of the chain
+        do not differ significantly regarding Geweke test -> Burn-In
+
+    """
+    # Get parameter samples as numpy arrays
+    chain = np.array(result.sample_result['trace_x'][0])
+
+    # Calculate burn in index
+    burn_in = burn_in_by_sequential_geweke(chain=chain,
+                                           zscore=zscore)
+    print('Geweke Burn-in index: '+str(burn_in))
+
+    result.sample_result['burn_in'] = burn_in
+
+    return result
+
+
+def chain_auto_correlation(result: Result):
+    """
+    Calculates the auto-correlation of the MCMC samples.
+
+    Parameters
+    ----------
+    result:
+        The pyPESTO result object with filled sample result.
+
+    Returns
+    -------
+    tau:
+        Array with the auto-correlation time tau for each parameter
+        dimension. We suggest taking the maximum over all components.
+
+    """
+
+    # Burn in index
+    burn_in = result.sample_result['burn_in']
```

Comment:
I think it would be good to set the burn_in in the result to `None` by default, and if it is None here, then raise an error that one has to compute it first via ... function.